### ISSUE TITLE
The X[~example 4]GB limit on `MPSTemporaryNDArray` is too restrictive for modern LLM workloads and prevents efficient use of PyTorch's MPS backend for inference tasks.

### ISSUE THREAD

**Issue by deep-learner-ConfigurableAI** [vllm-project/vllm-metal#43]
The vLLM Metal backend crashes during inference due to Metal Performance Shaders (MPS) framework limitation that prevents allocation of temporary NDArray buffers exceeding 4GB (2^32 bytes).

## Error Message

```
/AppleInternal/Library/BuildRoots/05cadd76-7a9b-11ef-9ad7-daac7d5d70b1/Library/Caches/com.apple.xbs/Sources/MetalPerformanceShaders/MPSCore/Types/MPSNDArray.mm:850: 
failed assertion `[MPSTemporaryNDArray initWithDevice:descriptor:isTextureBacked:] Error: total bytes of NDArray > 2**32'
```

## Environment

- **OS**: macOS (Apple Silicon)
- **Framework**: Metal Performance Shaders (MPS)
- **PyTorch Version**: With MPS backend support
- **Model**: Qwen/Qwen2.5-0.5B-Instruct (vocab_size ≈ 151,936)
- **Python**: 3.12

### What Happens

1. Model loads successfully using MLX on Metal/GPU
2. Model warm-up completes without issues
3. On first inference request, the model generates logits tensor with shape `(batch_size, vocab_size)`
4. When converting logits from MLX → PyTorch and moving to MPS device via `tensor.to('mps')`, Metal attempts to create temporary buffers
5. The temporary buffer allocation exceeds 4GB limit, causing Metal framework assertion failure
6. **The entire process crashes immediately** with the above error

Details:
- **Tensor Size**: For vocab_size=151,936 and batch_size=1 in float32:
  - Base size: 151,936 × 4 bytes = ~608 MB
  - However, Metal's internal operations may require additional temporary buffers
  - Memory fragmentation or multiple temporary allocations can push total beyond 4GB
- **MPS Limitation**: `MPSTemporaryNDArray` has a hardcoded limit of 2^32 bytes (4GB)

  - This is enforced at MPSNDArray.mm:850
  - No workaround or configuration option available

Here is How i fixed it : Both files (model_runner.py & tensor_bridge.py attached for reference)
**Workaround**:  

<img width="1721" height="1074" alt="Image" src="https://github.com/user-attachments/assets/f14d3f5f-1fc7-424b-a806-9aebce5d0c82" />
<img width="1717" height="1062" alt="Image" src="https://github.com/user-attachments/assets/85dbcac6-8958-437a-9969-a48b134326da" />

Forces CPU usage (loses GPU acceleration benefits)

[model_runner.py](https://github.com/user-attachments/files/24401114/model_runner.py)
[tensor_bridge.py](https://github.com/user-attachments/files/24401115/tensor_bridge.py)

**Comment by ericcurtin** [comment]
Can you post the vllm serve, client request, etc. to reproduce this?

**Comment by ericcurtin** [comment]
If we are confident on this fix, we should open a PR for review
